ts/components/messages/paginated/MessageList/Item.tsx

```diff
@@ -104,39 +104,89 @@ const styles = StyleSheet.create({
   }
 });
 
-function tagOrIcon({
+/**
+ * The data structure representing a badge.
+ */
+enum ItemBadge {
```

Comment:
We often prefer _string literals_ in place of _enum_
This is because _string literals_ are exhaustive when used in switch statements. In this way you can't forget a case at compile time

```suggestion
type ItemBadge = "paid" | "archived" | "qrcode";
```

i.e if a devs add another case
`type ItemBadge = "paid" | "archived" | "qrcode" | "anotherItem";`
the relative switch will raise an error (obviously you should omit the default case)

![Schermata 2022-01-31 alle 15 24 47](https://user-images.githubusercontent.com/822471/151810985-166c13cf-818a-4c27-b3c7-0b566af3be68.png)

what do you think? could it be an improvement?

Reply:
You have this behaviour with enums too! (or I'm missing something?)

![image](https://user-images.githubusercontent.com/5963683/151820040-483863fa-f25c-40c2-bd9a-c00d28633107.png)

But if in the code base string literals are the preferred approach there are no problems with me :) I just used enums instinctively, gonna change it in the next commit!

Reply:
You are right! Thank you! I forget that also enums could handle this case. 
Anyway I missed another important pain point with enum
Since enum, as you coded, are converted in a numeric const
![Schermata 2022-01-31 alle 16 27 20](https://user-images.githubusercontent.com/822471/151821707-d60be9e9-0c5b-49a7-92e8-bec4ddc1a1ae.png)

you can write this code without any errors at compile time
![Schermata 2022-01-31 alle 16 26 01](https://user-images.githubusercontent.com/822471/151821793-670ae523-a419-4ebf-a3ea-d4d681cddb71.png)

Reply:
Didn't know this could actually happen sometimes 😮 It's definitely a dangerous behaviour. I currently get a proper type-check, but if that could happen then yes, It's definitely dangerous!

![image](https://user-images.githubusercontent.com/5963683/151823011-c3833e07-a192-43b9-86fd-f1b274a46645.png)

I replaced the enum in the latest commit anyway 🙂

Reply:
You're right, the value it's opaque so that is actually accepted! I changed it in the latest commit 🙂